### ISSUE TITLE
Sanity check: MCAP self contained schema vs. OSI proto from dependency

### ISSUE THREAD

**Issue by jdsika** [OpenSimulationInterface/osi-utilities#4]
Make sure that the mcap contains the same message schema as the protobuffer that we use. How to proceed with downwards compatibitlity 3.7 reads/writes 3.3

**Comment by pmai** [comment]
There should be no strict checking: It is useful to read older OSI with newer OSI as well as newer OSI with older OSI (as long as major version matches); similarly it is seldom useful to write an older OSI version as you are using.

**Comment by TimmRuppert** [comment]
I agree with @pmai. It may be reasonable to read a message from an older 3.6 OSI of an MCAP file into a newer OSI 3.7 message format.

The user can inspect the channel metadata - specifically the stored OSI version - before deciding whether to proceed with reading the message or intentionally skip it.

Nonetheless, if you would like some kind of safety net, we could include such a check that issues a warning to `stderr`. This check could be enabled by default and should be easy to disable once, [similar to the existing option for skipping non-OSI messages.](https://github.com/lichtblick-suite/asam-osi-utilities/blob/main/include/osi-utilities/tracefile/reader/MCAPTraceFileReader.h#L57C1-L57C75)
